src/language-server/language-server.ts

```diff
@@ -101,33 +102,41 @@ export default class LanguageServer extends Handler {
     this.documents.listen(connection);
     this.converter = new AnalyzerLSPConverter(workspaceUri);
 
-    const synchronizer = new FileSynchronizer(
+    this.fileSynchronizer = new FileSynchronizer(
         connection, this.documents, workspacePath, this.converter);
 
-    this.settings = new Settings(connection, synchronizer, this.converter);
+    this.settings =
+        new Settings(connection, this.fileSynchronizer, this.converter);
     this.disposables.push(this.settings);
 
     this.editorService = new LocalEditorService({
-      urlLoader: synchronizer.urlLoader,
+      urlLoader: this.fileSynchronizer.urlLoader,
       urlResolver: new PackageUrlResolver(),
-      settings: this.settings
     });
+    this.analyzer = this.editorService.analyzer;
 
+    // Keep the analyzer up to date.
     this.disposables.push(
-        synchronizer.fileChanges.listen((filesChangeEvents) => {
+        this.fileSynchronizer.fileChanges.listen((filesChangeEvents) => {
           this.handleFilesChanged(filesChangeEvents);
         }));
-    this.fileSynchronizer = synchronizer;
 
     this.diagnosticGenerator = new DiagnosticGenerator(
-        this.editorService.analyzer, this.converter, connection, this.settings,
-        synchronizer, this.documents);
+        this.analyzer, this.converter, connection, this.settings,
+        this.fileSynchronizer, this.documents);
     this.disposables.push(this.diagnosticGenerator);
 
     const commandExecutor =
         new CommandExecutor(this.connection, this.diagnosticGenerator);
     this.disposables.push(commandExecutor);
 
+    const featureFinder = new FeatureFinder(this.analyzer);
+    const hoverDocumenter =
+        new HoverDocumenter(this.connection, this.converter, featureFinder);
+    this.disposables.push(hoverDocumenter);
```

Comment:
(opinable) you could avoid creating those 2 const above and inline everything here. Unless the plan is to use those const in some future PR

Reply:
The feature finder will be used in a bunch of places. 

For the second, I found it more legible to declare it and then add it as a disposable separately.